app/src/main/java/com/github/onlynotesswent/ui/common/Popups.kt

```diff
@@ -137,6 +149,108 @@ fun CreationDialog(
+  var folderSubFolders by remember { mutableStateOf<List<Folder>>(emptyList()) }
+  val userRootFolders = folderViewModel.userRootFolders.collectAsState()
+
+  Dialog(onDismissRequest = { onDismiss() }) {
+    Box(
+        modifier =
+            Modifier.testTag("FileSystemPopup")
+                .fillMaxWidth(0.9f) // Adjust the popup width
+                .fillMaxHeight(0.5f)
+                .padding(16.dp)
+                .background(
+                    color = MaterialTheme.colorScheme.background,
+                    shape = RoundedCornerShape(12.dp))) {
+          Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
+            Box(
+                modifier =
+                    Modifier.fillMaxWidth()
+                        .background(MaterialTheme.colorScheme.primary)
+                        .padding(vertical = 12.dp, horizontal = 16.dp)) {
+                  Text(
+                      text =
+                          if (selectedFolder == null) "Folders in: root"
+                          else "Folders in: ${selectedFolder!!.name}",
+                      style = MaterialTheme.typography.bodyMedium,
+                      color = MaterialTheme.colorScheme.onPrimary)
+                }
+
+            Column(
+                modifier =
+                    Modifier.weight(1f)
+                        .padding(start = 16.dp, end = 16.dp, top = 8.dp)
+                        .verticalScroll(rememberScrollState()),
+                verticalArrangement = Arrangement.spacedBy(4.dp)) {
+                  if (selectedFolder == null) {
+                    userRootFolders.value.forEach { folder ->
+                      Box(
+                          modifier =
+                              Modifier.testTag("FileSystemPopupFolderChoiceBox" + folder.id)
+                                  .fillMaxWidth()
+                                  .background(
+                                      color = MaterialTheme.colorScheme.surface,
+                                      shape = RoundedCornerShape(8.dp))
+                                  .clickable {
+                                    folderViewModel.getSubFoldersOfNoStateUpdate(
+                                        folder.id,
+                                        onSuccess = { subFolders -> folderSubFolders = subFolders })
+                                    selectedFolder = folder
+                                  }
+                                  .padding(12.dp)) {
+                            Text(
+                                modifier = Modifier.testTag("FileSystemPopupFolderChoiceText"),
+                                text = folder.name,
+                                style = MaterialTheme.typography.bodyMedium,
+                                color = MaterialTheme.colorScheme.onSurface)
+                          }
+                    }
+                  } else {
+                    folderSubFolders.forEach { subFolder ->
```

Comment:
Maybe since both statements in the if and else are quite similar, extracting them as a separated function could be done

Reply:
I left it as is as in the second PR I think I'll have a different content depending if we are at Root

Reply:
Understood, no problem